packages/iris-grid/src/IrisGridModelUpdater.tsx

```diff
@@ -45,159 +45,153 @@ interface IrisGridModelUpdaterProps {
 /**
  * React component to keep IrisGridModel in sync
  */
-const IrisGridModelUpdater = React.memo(
-  ({
-    model,
-    modelColumns,
-    top,
-    bottom,
-    left,
-    right,
-    filter,
-    formatter,
-    reverseType = TableUtils.REVERSE_TYPE.NONE,
-    sorts,
-    customColumns,
-    movedColumns,
-    hiddenColumns,
-    alwaysFetchColumns,
-    rollupConfig = null,
-    totalsConfig = null,
-    selectDistinctColumns = EMPTY_ARRAY,
-    pendingRowCount = 0,
-    pendingDataMap = EMPTY_MAP,
-    frozenColumns,
-    formatColumns,
-    columnHeaderGroups,
-    partitionConfig,
-  }: IrisGridModelUpdaterProps) => {
-    const columns = useMemo(
-      () =>
-        IrisGridUtils.getModelViewportColumns(
-          modelColumns,
-          left,
-          right,
-          movedColumns,
-          hiddenColumns,
-          alwaysFetchColumns,
-          COLUMN_BUFFER_PAGES
-        ),
-      [
-        modelColumns,
+function IrisGridModelUpdater({
+  model,
```

Comment:
Don't remove the `React.memo`. It is the equivalent to pure components in class based react and means if the props don't change then the component does not re-render

Reply:
I suggested he remove it. Problem is columns can change without the model changing. I think it's fine if this re-renders as it's going to be checking `useOnChange` anyway before applying changes. https://github.com/deephaven/web-client-ui/pull/2183#discussion_r1718831523

Reply:
Still seems like a smell to me. We're getting rid of the memoization by React to memoize it based on props ourselves instead? I think the smell is this doesn't belong on formatter which then indirectly changes the visible columns. Formatter prrior to this should only change things when it's directly called. Now it's indirectly changing the visible columns instead of just setting that directly in the model